packages/components/src/components/Modal/Modal.test.tsx

```diff
@@ -7,79 +7,84 @@ const OPEN_MODAL_TEXT = "Open modal";
 
 describe("<Modal />", () => {
   it("should open and close a modal using the close modal button", async () => {
+    const user = userEvent.setup();
     render(<Default />);
     const renderedOpenButton = screen.getByText(OPEN_MODAL_TEXT);
     expect(renderedOpenButton).toBeInTheDocument();
 
-    await userEvent.click(renderedOpenButton);
+    await act(() => user.click(renderedOpenButton));
     expect(screen.getByRole("dialog")).toBeInTheDocument();
     expect(screen.getByRole("heading", { level: 2 })).toBeInTheDocument();
 
     const renderedCloseButton = screen.getByLabelText("Close modal");
     expect(renderedCloseButton).toBeInTheDocument();
 
-    await userEvent.click(renderedCloseButton);
+    await act(() => user.click(renderedCloseButton));
     expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
   });
 
   it("should open and close a modal using the primary button", async () => {
+    const user = userEvent.setup();
     render(<Default />);
     const renderedOpenButton = screen.getByText(OPEN_MODAL_TEXT);
     expect(renderedOpenButton).toBeInTheDocument();
 
-    await userEvent.click(renderedOpenButton);
+    await act(() => user.click(renderedOpenButton));
     expect(screen.getByRole("dialog")).toBeInTheDocument();
     expect(screen.getByRole("heading", { level: 2 })).toBeInTheDocument();
 
     const renderedPrimaryButton = screen.getByText("Done");
     expect(renderedPrimaryButton).toBeInTheDocument();
 
-    await userEvent.click(renderedPrimaryButton);
+    await act(() => user.click(renderedPrimaryButton));
     expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
   });
 
   it("should open and close a modal using the secondary button", async () => {
+    const user = userEvent.setup();
```

Comment:
Nit: we could move this line to the start of the test and avoid having to repeat it in multiple test cases